### ISSUE TITLE
Success + Error states on Create + Edit Groups page

### ISSUE THREAD

**Issue by zakhap** [hicommonwealth/commonwealth#5150]
## Description

Further design additions to the work on #4991. Nothing fundamentally changes, but this additional ticket was created to detail the additional UI affordances Jessica added after the original ticket was handed off.

DevMode Link:
https://www.figma.com/file/ePuGFatYQI6GL72i27sVwj/Gating?type=design&node-id=1221-31750&mode=dev

## Error State Requirements, Form Behavior

- All Behavior details are listed next to the individual form fields in the figma. 
- Important to note: Every single input field has some sort of validation + error state. 
- Examples of how it looks in the figma are below in the addition context.

## Additional context

<img width="694" alt="Screen Shot 2023-09-25 at 6 07 29 PM" src="https://github.com/hicommonwealth/commonwealth/assets/16248081/b1224127-04bb-483f-80d2-3afc01c55795">
<img width="697" alt="Screen Shot 2023-09-25 at 6 09 51 PM" src="https://github.com/hicommonwealth/commonwealth/assets/16248081/ef220073-5055-4e4c-b349-a20a9a770fee">
<img width="727" alt="Screen Shot 2023-09-25 at 6 09 55 PM" src="https://github.com/hicommonwealth/commonwealth/assets/16248081/ad1218b3-8803-4cbe-9b23-6d13cb967715">

**Comment by mzparacha** [comment]
Will do this as part of https://github.com/hicommonwealth/commonwealth/pull/5045

**Comment by mzparacha** [comment]
With zod, our validation schema (simplified example) looks like this
```
const schema = z.object({
  requirementType: z
    .string({ invalid_type_error: 'Type is required!' })
    .nonempty({ message: 'Type is required!' }),
  requirementContractAddress: z
    .string({ invalid_type_error: 'Address is required!' })
    .nonempty({ message: 'Address is required!' })
});
```
and object validations will looks like this
```
const obj1 = { requirementType: 'erc-20', requirementContractAddress: 'abchdvhcdvshchvcgdccghdghds'}
schema.parse(obj1) // success

const obj2 = { requirementType: 'erc-20', requirementContractAddress: ''}
schema.parse(obj2) // failure: requirementContractAddress is required

const obj3 = { requirementType: 'erc-20'}
schema.parse(obj3) // failure: requirementContractAddress is required

// when requirement type has a 'token' keyword in it then we don't want to allow requirementContractAddress, but right not it fails with "requirementContractAddress is required" error
const obj4 = { requirementType: 'evm-token'}
schema.parse(obj4) // failure: requirementContractAddress is required

const obj5 = { requirementType: ''}
schema.parse(obj5) // failure: requirementType is required, requirementContractAddress is required <- notice it will throw these as 2 separate errors
```

and we want to achieve this (basically modify the schema to support all this)
```
// 1. When requirement type has a value that include's the keyword 'token', we want the requirementContractAddress to be optional, otherwise its required
// i.e schema.parse should work like this
const obj1 = { requirementType: 'cosmos-token', requirementContractAddress: ''}
schema.parse(obj1) // success
const obj2 = { requirementType: 'evm-token', requirementContractAddress: ''}
schema.parse(obj2) // success
const obj3 = { requirementType: 'erc-20', requirementContractAddress: ''}
schema.parse(obj3) // failure: requirementContractAddress is required
const obj4 = { requirementType: 'erc-20'}
schema.parse(obj4) // failure: requirementContractAddress is required

// 2. When requirement type has no value, we dont want to validate any other field (other than requirementType)
// i.e schema.parse should work like this
const obj1 = { requirementType: ''}
schema.parse(obj1) // failure: requirementType is required <- currently it will throw 2 separate errors (the other one is "requirementContractAddress is required"), but it should only throw the one specified i.e only validate other fields if requirementType has a value
const obj2 = { requirementType: 'evm-token'}
schema.parse(obj2) // success
const obj2 = { requirementType: 'evm-token', requirementContractAddress: ''}
schema.parse(obj2) // failure: requirementContractAddress is not allowed
const obj2 = { requirementType: 'erc-20'}
schema.parse(obj2) // failure: requirementContractAddress is required
const obj2 = { requirementType: 'erc-20', requirementContractAddress: 'z4wxe5cr6vt7by8nu9mi'}
schema.parse(obj2) // success
```

cc: @jnaviask @masvelio @rbennettcw

**Comment by rbennettcw** [comment]
Ok, so this is related to the gating API.

The backend already performs validation of the requirements. It uses Zod to validate the general request shape, but uses the `ajv` library to validate requirements– based on a JSON schema.

My recommendation is to let only backend handle requirements validation. The response payload includes validation error details that you can parse and use to update the UI.

If we perform validation on both the frontend and backend, then we'll have to keep both of them in sync, which is less maintainable.

**Comment by mzparacha** [comment]
I agree with the thought process here, but the issue is actually related to how we validate each field.

1. Currently, we validate each field as the user types/selects or interacts with the field. Considering a user is typing 10 chars, that would be 10 API calls.

2. We also updated our forms strategy https://github.com/hicommonwealth/commonwealth/issues/4150 to work based on the provided validation schema, so just a simple validation schema of any validation library supported by [react-hook-forms](https://react-hook-form.com/) will work on our forms. Also, this makes it easier to do a lot of validations without any react state (which is very good). To sum this up: the validation schema provides our form output i.e error messages, and sanitized values in b/w are provided by the schema we use.

Maybe we can come up with a general way to manage both or abstract them. Also, in some cases, frontend validation will be different than the one from the backend in terms of object shape/keys/validation (messages/types), etc.

cc: @jnaviask @rbennettcw @masvelio

**Comment by masvelio** [comment]
It really depends on how design/product wants the UX for form validation should look like. From my experience, backend validation is always non negotiable. For frontend, it depends:
1.  If we want to have "real time" validation, eg `onInputChange`, then we have to go with frontend validation. Otherwise, as @mzparacha mentioned, we would need to call validationApi every key stroke
2. Other way is that we have validation once the input is blurred OR once the whole form is submitted. In my opinion, it is much worse from UX standpoint.

My take on this is that we should both have front end and back end validation (solution 1). If front and api uses the same schema, maybe it would be wise to abstract it out to the common module.

**Comment by CowMuon** [comment]
Fix conflicts on PR pls and let's get Gating on Frack.
